.gitlab-ci.yml

```diff
@@ -58,6 +58,26 @@ variables:
   environment:
     name: parity-build
 
+.build_to_own_reg:                 &docker_build_to_own_reg
+  <<:                              *docker_build
+  before_script:
+    - test -z "$DOCKERIMAGE" && echo "DOCKERIMAGE must be defined" && exit 1
+    - docker login -u $CI_REGISTRY_USER -p $CI_REGISTRY_PASSWORD $CI_REGISTRY
```

Comment:
would be cool to avoid this kind of login somehow

Reply:
these are built-in Gitlab variables that change every job actually, AFAIU it's the most secure way to login to gitlab registry in order to `push` into it.

Reply:
nice :+1: